File: demolayout/src/main/java/com/tution/demolayout/MainActivity.java
package com.tution.demolayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.constraintlayout.widget.ConstraintSet;

import android.os.Bundle;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);

        ConstraintLayout constraintLayout = new ConstraintLayout(this);

        Button button = new Button(this);
        button.setText("ok");
        constraintLayout.addView(button);

        Button button1 = new Button(this);

        constraintLayout.addView(button1);
        ConstraintSet set = new ConstraintSet();

        set.connect(button.getId(),ConstraintSet.BOTTOM,ConstraintSet.PARENT_ID,
                ConstraintSet.BOTTOM,12);
        set.constrainHeight(button.getId(), ConstraintSet.WRAP_CONTENT);
        set.constrainWidth(button1.getId(), ConstraintSet.WRAP_CONTENT);




    }
}

File: kotlin/src/main/java/aop/demo/jetpack/android/kotlin/MainActivity.kt
package aop.demo.jetpack.android.kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.constraintlayout.motion.widget.MotionLayout
import androidx.core.net.toUri
import aop.demo.jetpack.android.kotlin.impl.repository.model.LoginBean
import aop.demo.jetpack.android.kotlin.impl.ui.ui.login.LoginActivity
import org.jetbrains.anko.AnkoLogger
import org.jetbrains.anko.info
import java.lang.Exception
import kotlin.coroutines.resume
import kotlin.coroutines.resumeWithException
import kotlin.coroutines.suspendCoroutine

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_scene1)
//        val intent = Intent(this, LoginActivity::class.java)
//        startActivity(intent)

    }


    suspend fun getUserCoroutine() = suspendCoroutine<LoginBean> {
        continuation ->
        continuation.resume(LoginBean(false, listOf(), listOf(),"7777","dafa",121341,"fadsfas",
                "fdasfs","fdsafdasf",1,"fdasfdas"))
        continuation.resumeWithException(exception = Exception("something wrong"))
    }
}
